src/REPL/Repl.cpp

```diff
@@ -8,54 +13,78 @@ namespace Ark
     
     void Repl::run()
     {
-        const std::string new_prompt = ">> ";
-        const std::string continuing_prompt = ".. ";
+        Ark::State state(m_options, m_lib_dir);
+        Ark::VM vm(&state);
+        state.setDebug(0);
+        std::string code;
+        bool init = false;
 
         print_repl_header();
+        cgui_setup();
 
-        while (true)
+        while(true)
         {
-            Ark::State state(m_options, m_lib_dir);
-            Ark::VM vm(&state);
-            state.setDebug(0);
-
-            std::stringstream code;
-            int open_parentheses = 0;
-            int open_braces = 0;
-            bool new_command = true;
+            std::stringstream tmp_code;
+            unsigned open_parentheses = 0;
+            unsigned open_braces = 0;
 
-            std::cout << new_prompt;
-            for(std::string line; std::getline(std::cin, line);)
+            tmp_code << code;
+            while(true)
             {
-                trim_whitespace(line);
-
-                // empty lines and comments handling
-                if (line.length() == 0 || line.at(0) == '#')
+                std::string str_lines = "000";
+                if(std::to_string(m_lines).size() < 3)
                 {
-                    std::cout << (new_command ? new_prompt : continuing_prompt);
-                    continue;
+                    std::size_t size = std::to_string(m_lines).size(); 
+                    str_lines.replace((str_lines.size() - size), size, std::to_string(m_lines));
                 }
+                else
+                    str_lines = std::to_string(m_lines);
+
+                std::string infos = "main:" + str_lines + "> ";
+                std::string line = m_repl.input(infos);
+
+                // line history
+                m_repl.history_add(line);
+                trim_whitespace(line);
 
                 // specific commands handling
-                if (line.compare("(quit)") == 0)
+                if(line == "(quit)")
```

Comment:
`if (line == "(quit)" || line == "(q)")` since (q) can also quit the REPL

Reply:
I think only "(qui)" is better

Reply:
then the header of the repl should be changed to mention only (quit) :D

Reply:
Yes